src/m365/spo/commands/tenant/tenant-site-list.spec.ts

```diff
@@ -97,11 +97,21 @@ describe(commands.TENANT_SITE_LIST, () => {
     assert.notStrictEqual(actual, true);
   });
 
+  it('fails validation if withOneDriveSites is specified together with the type option', async () => {
+    const actual = await command.validate({ options: { type: 'TeamSite', withOneDriveSites: '1' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
   it('fails validation if includeOneDriveSites is specified together with the webTemplate option', async () => {
     const actual = await command.validate({ options: { webTemplate: 'STS#3', includeOneDriveSites: '1' } }, commandInfo);
     assert.notStrictEqual(actual, true);
   });
 
+  it('fails validation if withOneDriveSites is specified together with the webTemplate option', async () => {
```

Comment:
let's remove the duplicated tests with the old option name and add a test to check if deprecation waring message is present